docs(participate): clarify delegator withdrawal delay and slashing impact (#23695)
AztecProtocol/aztec-packages: docs/docs-participate/token/delegation.md

displayed_sidebar: participateSidebar
---

# Delegating stake

If you want to participate in staking but don't want to run your own infrastructure, you can delegate your tokens to professional operators who run sequencers on your behalf.

## Before you delegate

Understanding these concepts will help you choose the right operator:

- [Staking tokens](/participate/token/staking): how proof of stake works
- [Economics and rewards](/participate/token/economics): how rewards are distributed
- [How governance works](/participate/governance): understand voting power

## How delegation works

When you delegate tokens to an operator:

1. **Your tokens are staked** through the operator's sequencer
2. **The operator runs infrastructure** on your behalf
3. **Rewards are shared** between you and the operator based on their fee structure

> 4. **Slashing risk falls on your stake**: each attester is funded by exactly one Token Vault. If the operator runs the node poorly and the attester is slashed, the burn comes out of your Token Vault's position. The operator does not contribute tokens to the same attester and so does not share the financial loss directly.

Reviewer: One nit is not all delegators use Token Vaults. Maybe use "comes out of your balance?"

Author: updated throughout the doc, replacing "token vault" with "staking position" in most places.

For how unstaking and slashing affect a delegator's balance specifically, see [Withdrawing delegated stake](#withdrawing-delegated-stake) and [What happens if your operator is slashed](#what-happens-if-your-operator-is-slashed) below.

## Choosing an operator

When selecting an operator to delegate to, consider:

### Performance metrics

- **Uptime**: how reliably does the operator maintain their infrastructure?
- **Attestation rate**: do they consistently participate in consensus?
- **Slashing history**: have they been slashed before?

### Economic terms

- **Commission rate**: what percentage of rewards does the operator keep?

> - **Minimum delegation**: is there a minimum amount required?

Reviewer: there is no minimum delegation set by the providers today

### Reputation

- **Track record**: how long have they been operating?
- **Community standing**: are they known in the Aztec community?
- **Transparency**: do they communicate openly about their operations?

## Delegation process

### Prerequisites

- An Ethereum wallet that owns an Aztec Token Vault (the same wallet you connected during the token sale)
- A Token Vault balance of at least 200,000 AZTEC tokens

:::note Minimum stake

> You must stake a minimum of 200,000 AZTEC per validator, similar to how Ethereum requires 32 ETH per validator.

Reviewer: It's not actually like Ethereum because you can stake more than 32ETH on Ethereum. The 200k is not a minimum, it's a fixed constant.

You cannot consolidate multiple Token Vaults into a single delegation. Each vault must be staked individually.
:::

### Step 5: set operator address

> The operator address controls block submissions for this vault. Confirm the address is correct, as this determines who manages sequencer operations and receives reward attribution.

Reviewer: ```suggestion
The operator address controls sequencer operations for this vault. Confirm the address is correct, as this address can choose who to delegate to and pick the reward attribution address
```

Reviewer: FYI that the frontend does not allow them to set an "operator" address. The operator always == beneficiary for the frontend.

:::

### Step 7: approve tokens

Approve the staker contract to move funds from your Token Vault. Each validator requires 200,000 tokens, so the approval amount matches your delegation.

### Step 8: delegate

Review your delegation configuration and click **Delegate** / **Add to Batch**.

### Step 9: execute batch

Review the full set of queued transactions and click **Execute All**.

![Batch execution review screen](/img/batch-execute.png)

:::tip Subsequent delegations
When you delegate again from a vault that already has an operator and staker version configured, Steps 5 and 6 are skipped, making future delegations faster.
:::

## Managing your delegation

### Monitoring performance

Keep track of your delegated stake:
- Check operator uptime and performance
- Monitor for any slashing events
- Review reward distributions

### Changing operators

If you want to switch to a different operator:
1. Initiate undelegation from your current operator
2. Wait for the unbonding period
3. Delegate to your new chosen operator

## Withdrawing delegated stake

> When you initiate withdrawal of delegated stake from the dashboard, the same exit timing applies as for self-staking: roughly **38 days on mainnet** and **2 days on testnet** before you can finalize. The dashboard shows the exact unlock time for your position.

Reviewer: Same comment (up to you though)

The reason: delegated stake reaches the same Governance Staking Escrow (GSE) as self-staked tokens, so the governance withdrawal delay (`votingDelay/5 + votingDuration + executionDelay`) applies in both cases. See [Staking tokens, exit delays](/participate/token/staking#exit-delays) for the full breakdown of the two concurrent delays and why the governance delay applies to every exit.

> If your operator is slashed during the exit window, your share of the pool is reduced proportionally before you can finalize the withdrawal. There is no way to exit faster after a slash.

Reviewer: I wouldn't mention "pool"

## What happens if your operator is slashed

Each attester (validator key) is funded by exactly one Token Vault holding the 200,000-token activation threshold. When you delegate, your Token Vault funds that attester; the operator runs the node on your behalf but does not contribute tokens to the same attester. Slashing penalties for that attester therefore come entirely out of the position your Token Vault funded.

> - Each slashable offense reduces the attester's stake by the per-offense penalty configured at the protocol level. Once the remaining stake falls below the configured ejection threshold, the attester is automatically removed from the validator set. The current per-offense penalties and the ejection threshold are listed in [Slashing and offenses](/operate/operators/sequencer-management/slashing_and_offenses).

Reviewer: I think we removed the exact penalties from the linked doc.

Author: i am updating the wording
